doc/user/methods/gradient_boosting.md

```diff
@@ -0,0 +1,223 @@
+
+ * `gb.Train(data, labels, numClasses, numWeakLearners)`
+   - Train on numerical data without using any decision tree arguments.
+
+---
+
+ * `gb.Train(data, labels, numClasses, numWeakLearners, minimumLeafSize, minimumGainSplit, maximumDepth)`
+   - Train on numerical data using weak learner model's arguments `minimumLeafSize`, `minimumGainSplit` and `maximumDepth`.
+
+---
+
+Types of each argument are the same as in the table for constructors
+[above](#constructor-parameters).
+
+***Note:***: Training is not incremental.  A second call to `Train()` will retrain the
+model from scratch.
+
+### Classification
+
+Once a `GradBoosting` object is trained, the `Classify()` member function can be used
+to make class predictions for new data.
+
+ * `size_t predictedClass = gb.Classify(point)`
+    - ***(Single-point)***
+    - Classify a single point, returning the predicted class.
+
+---
+
+ * `gb.Classify(point, prediction)`
+    - ***(Single-point)***
+    - Classify a single point and compute class probabilities.
+    - The predicted class is stored in `prediction`.
+
+---
+
+ * `gb.Classify(data, predictedLabels)`
+    - ***(Multi-point)***
+    - Classify a set of points.
+    - The prediction for data point `i` can be accessed with `predictedLabels[i]`.
+
+---
+
+#### Classification Parameters:
+
+| **usage** | **name** | **type** | **description** |
+|-----------|----------|----------|-----------------|
+| _single-point_ | `point` | [`arma::vec`](../matrices.md) | Single point for classification. |
+| _single-point_ | `prediction` | `size_t&` | `size_t` to store class prediction into. |
+||||
+| _multi-point_ | `data` | [`arma::mat`](../matrices.md) | Set of [column-major](../matrices.md#representing-data-in-mlpack) points for classification. |
+| _multi-point_ | `predictedLabels` | [`arma::Row<size_t>&`](../matrices.md) | Vector of `size_t`s to store class prediction into.  Will be set to length `data.n_cols`. |
+
+***Note:*** different types can be used for `data` and `point` (e.g.
+`arma::fmat`, `arma::sp_mat`, `arma::sp_vec`, etc.).  However, the element type
+that is used should be the same type that was used for training.
+
+### Other Functionality
+
+ * `WeakLearner(size_t i)` returns a `WeakLearnerType` indicating which weak learner 
+    is being used. 
+
+ * `gb.NumWeakLearners()` returns a `size_t` indicating how many weak learners are being
+    used by the model.
+
+ * `gb.NumClasses()` returns a `size_t` indicating the number of classes the
+   model was trained on.
+
+ * `gb.SetNumWeakLearners(size_t x)` sets the number of weak learners to `x` explicitly.
+
+For complete functionality, the [source
+code](/src/mlpack/methods/grad_boosting/grad_boosting.hpp) can be consulted.
+Each method is fully documented.
+
+### Simple Examples
+
+See also the [simple usage example](#simple-usage-example) for a trivial use of
+`GradBoosting`.
+
+---
+
+Train a gradient boosting model on mixed categorical data and save it:
```

Comment:
I missed this on the first review: `GradBoosting` does not support mixed categorical data?  This example is actually insidiously wrong: it loads mixed categorical data, but then the `info` object is *ignored* when we call `Train()`.  So what actually happens in this example is that we load mixed categorical data, but then train a model, treating every dimension as a numeric type.  This is really misleading, and is a great example of why you cannot simply copy examples from other methods and adapt without thinking.

Either we should add support for categorical data in this PR, or, we should adapt this example to use a numerical-only dataset.

Reply:
Sorry about that, it was running on my system, so I didn't realize this. I'll add categorical support ASAP

Reply:
Yeah, it will run just fine (and probably produce a good model too given the nature of the covertype dataset), but it subtly doesn't do what it says the example does.  Sounds good about the categorical support, thanks! :+1: